sql/parser/eval.go

```diff
@@ -396,6 +398,21 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (DBool, error){
 	cmpArgs{LE, intervalType, intervalType}: func(left Datum, right Datum) (DBool, error) {
 		return DBool(left.(DInterval).Duration <= right.(DInterval).Duration), nil
 	},
+
+	cmpArgs{Like, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
+		pattern := regexp.QuoteMeta(string(right.(DString)))
+		pattern = strings.Replace(pattern, "%", ".*", -1)
+		pattern = strings.Replace(pattern, "_", ".", -1)
+		pattern = fmt.Sprintf("^%s$", pattern)
+		fmt.Println(pattern)
+
+		re, err := regexp.Compile(pattern)
```

Comment:
[path.Match()](http://golang.org/pkg/path/#Match) implements shell-style pattern matching, which is closer to the capabilities of `LIKE` expressions than regexps, and it doesn't require compilation. On the other hand, it doesn't have something like `QuoteMeta` so it would be more difficult to transform the input string safely, and it's unclear how it performs compared to a regexp.